packages/flutter/lib/src/widgets/visibility.dart

```diff
@@ -0,0 +1,244 @@
+///  * [AnimatedCrossFade], which can fade between two specific children.
+class Visibility extends StatelessWidget {
+  /// Control whether the given [child] is [visible].
+  ///
+  /// The [child] and [replacement] arguments must not be null.
+  ///
+  /// The boolean arguments must not be null.
+  ///
+  /// The [maintainSemantics] and [maintainInteractivity] arguments can only be
+  /// set if [maintainSize] is set.
+  ///
+  /// The [maintainSize] argument can only be set if [maintainAnimation] is set.
+  ///
+  /// The [maintainAnimation] argument can only be set if [maintainState] is set.
+  const Visibility({
+    Key key,
+    @required this.child,
+    this.replacement = const SizedBox.shrink(),
+    this.visible = true,
+    this.maintainState = false,
+    this.maintainAnimation = false,
+    this.maintainSize = false,
+    this.maintainSemantics = false,
+    this.maintainInteractivity = false,
+  }) : assert(child != null),
+       assert(replacement != null),
+       assert(visible != null),
+       assert(maintainState != null),
+       assert(maintainAnimation != null),
+       assert(maintainSize != null),
+       assert(maintainState == true || maintainAnimation == false, 'Cannot maintain animations if the state is not also maintained.'),
+       assert(maintainAnimation == true || maintainSize == false, 'Cannot maintain size if animations are not maintained.'),
+       assert(maintainSize == true || maintainSemantics == false, 'Cannot maintain semantics if size is not maintained.'),
+       assert(maintainSize == true || maintainInteractivity == false, 'Cannot maintain interactivity if size is not maintained.'),
+       super(key: key);
+
+  /// The widget to show or hide, as controlled by [visible].
+  ///
+  /// {@macro flutter.widgets.child}
+  final Widget child;
+
+  /// The widget to use when the child is not [visible], assuming that none of
+  /// the `maintain` flags (in particular, [maintainState]) are set.
+  ///
+  /// The normal behavior is to replace the widget with a zero by zero box
+  /// ([SizedBox.shrink]).
+  ///
+  /// See also:
+  ///
+  ///  * [AnimatedCrossFade], which can animate between two children.
+  final Widget replacement;
+
+  /// Switches between showing the [child] or hiding it.
+  ///
+  /// The `maintain` flags should be set to the same values regardless of the
+  /// state of the [visible] property, otherwise they will not operate correctly
+  /// (specifically, the state will be lost regardless of the state of
+  /// [maintainState] whenever any of the `maintain` flags are changed, since
+  /// doing so will result in a subtree shape change).
+  ///
+  /// Unless [maintainState] is set, the [child] subtree will be disposed
+  /// (removed from the tree) while hidden.
+  final bool visible;
+
+  /// Whether to maintain the [State] objects of the [child] subtree when it is
+  /// not [visible].
+  ///
+  /// Keeping the state of the subtree is expensive. It should only be
```

Comment:
Would be a nice place to say why it's expensive if the reader would think at this point that it's not rebuilding most of the time, it's not laying out and it's not painting. 

Also, I'm thinking how do we balance the relative weight of the word 'expensive' we use around the codebase (for instance relative to `This effect is relatively expensive` on BackdropFilter which out of thin air I'm guesstimating that it's more expensive than this)

Reply:
i've tried to clarify this further